### PR TITLE
Identity facade for x/hybrid using sackmesser

### PR DESCRIPTION
**Pull request by danistrebel**
What's changed, or what was fixed?

- Identity facade support for hybrid/X (keeping Edge support)
- Deployment based on Sackmesser

- [x] I have run all the tests locally and they all pass.
- [x] I have followed the relevant style guide for my changes.

**CC:** @apigee-devrel-reviewers

### CODE REVIEW

File: references/identity-facade/pipeline.sh [hunk 1]
```diff
@@ -342,43 +240,97 @@ set_functional_test_env_var() {
...
+    # generate edge.json file
+    generate_edge_json "$APIGEE_ENV"
+
+    set_functional_test_env_var "$timestamp"
+
+    # deploy Apigee artifacts: proxy, developer, app, product cache, kvm and proxy
+    sackmesser deploy --apigeeapi -o "$APIGEE_ORG" -e "$APIGEE_ENV" -u "$APIGEE_USER" -p "$APIGEE_PASS" -d "$SCRIPTPATH"
+
+    # set developer app (apigee_client) credentials
+    curl --silent -X POST \
```
Comment (danistrebel) [inline references/identity-facade/pipeline.sh:271]: --fail

File: references/identity-facade/pipeline.sh [hunk 2]
```diff
@@ -342,43 +240,97 @@ set_functional_test_env_var() {
...
-set_devapp_credentials
+    set_functional_test_env_var "$timestamp"
 
-# set developer app (apigeee_client) product
-set_devapp_product
+    # # deploy Apigee artifacts: proxy, developer, app, product cache, kvm and proxy
+    sackmesser deploy --googleapi -o "$APIGEE_X_ORG" -e "$APIGEE_X_ENV" -t "$APIGEE_TOKEN" -h "$APIGEE_X_HOSTNAME" -d "$SCRIPTPATH"
+
+    # set developer app (apigee_client) credentials
+    curl --silent -X POST \
```
Comment (laughingbiscuit) [inline references/identity-facade/pipeline.sh:314]: --fail

File: references/identity-facade/pipeline.sh [hunk 3]
```diff
@@ -342,43 +240,97 @@ set_functional_test_env_var() {
...
-    echo "no tests run for custom OIDC Idp: $TEST_IDP_DISCOVERY_DOCUMENT"
+    # execute integration tests only against mock IDP
+    if [ -z ${IDP_DISCOVERY_DOCUMENT+x} ]; then
+        (cd "$SCRIPTPATH" && npm i && TEST_HOST="$APIGEE_X_HOSTNAME" npm run test)
+    else
+        echo "no tests run for custom OIDC Idp: $DP_DISCOVERY_DOCUMENT"
+    fi
 fi
+
+
```
Comment (laughingbiscuit) [inline references/identity-facade/pipeline.sh:333]: _trivial_: whitespace

**Review by laughingbiscuit — APPROVED** [review]
lgtm - thanks daniel

**Pull request merged**